libraries/tracker-core/src/payload.ts

```diff
@@ -28,7 +28,8 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import { base64urlencode } from './base64';
+import { base64urlencode, base64urldecode } from './base64';
+import type { SelfDescribingJson } from './core';
```

Comment:
I think this is introduced in TypeScript 3.8? It seems unlikely to cause any issues, but might be worth ensuring this doesn't break anything for users running earlier versions of typescript. I don't think it will, as its just an inport for this file, but I'm always nervous about this, it's way too easy to do (and I've definitely done it before in an earlier release by making some API function have some fancy TS syntax)

Reply:
It was introduced in TS 3.8, but I wasn't able to find out by googling how it is handled in previous versions. But I can't import from the core file without the `import type` directive as that would be a circular dependency (core also imports this file). So don't see how else to solve it (other than refactoring the files and extracting `SelfDescribingJson` to another file.

Reply:
This should be fine, I'm just nervous we'll break the TS definitions as part of the build output for earlier TS versions. It seems unlikely this import will make it into the final output though.

I think the quick check is to build the tracker package, load it via file reference into a react app with TS 3.7, import and initialise the tracker, and see if it compiles and sends an event.